core/trino-main/src/test/java/io/trino/sql/planner/iterative/rule/test/RuleAssert.java

```diff
@@ -159,6 +168,28 @@ public void matches(PlanMatchPattern pattern)
             assertPlan(session, metadata, ruleApplication.statsProvider, new Plan(actual, types, StatsAndCosts.empty()), ruleApplication.lookup, pattern);
             return null;
         });
+
+        return new WarningsMatcher(warningCollector.getWarnings());
```

Comment:
are the warnings cleared now from the collector?

(this is how i understand the point in returning `WarningsMatcher` directly from the `matches` or `doesNotFire`)

Reply:
Warnings are not cleared from the collector. I picked this way as it seemed cleaner than sending list of expected warnings into `matches` or `doesNotFire`.